src/api/student/student.service.ts

```diff
@@ -106,4 +107,112 @@ export class StudentService extends APIService<
 
     return studentPathInstance;
   }
+
+  async finishExams(user: Payload, createUserDto: FinishStudentQuiz): Promise<QuizInstance> {
+    const { fullMark, mark, answeredOptions, quizzId } = createUserDto;
+    const quizInstance = await this.db.quizInstance.findUnique({
+      where: {
+        id: quizzId,
+      },
+      include: {
+        quizStudents: true,
+      },
+    });
+    const quizStudent = quizInstance?.quizStudents.find((student) => student.studentId === user.sub);
+
+    if (!quizStudent) {
+      throw new Error('Quiz student not found');
+    }
+    await this.db.quizInstanceStudent.update({
```

Comment:
I think we need the https://www.prisma.io/docs/orm/reference/prisma-client-reference#upsert
+ the id is autoincrement